Feature/testimonial component

jhossyd3v opened the pull request:
Testimonial component
- In Home page the green shape is an image on mobile and created with CSS on desktop
- In Hypophosphatasia page the green shape is created with CSS for mobile and desktop
- Examples of testimonial in Home and Hypophosphatasia pages are in Design Page

Commit:
feat: :construction: Add files for Testimonial component

Add .astro file and.scss file for Testimonial component

Commit:
style: :bento: Add images for Homepage Testimonial

Add main image and shape image for homepage testimonial

Commit:
style: :lipstick: Add import for testimonial .scss and add color styles

Add import for testimonial scss in styles and add classes for required colors in utils

Commit:
style: :construction: Add Testimonial component in index

Add component on index page to build its styles and test

Commit:
Merge branch 'develop' of https://github.com/HackMort/strensiq-hcp into feature/testimonial-component

Commit:
style: :lipstick: Fix styles for Homepage testimonial

Fix styles for homepage testimonial on bigger screens

Commit:
style: :bento: Update assets for Testimonial component

Change the folder structure for Testimonial assets, create a testimonial folder inside pages folder where is required and standard names for images

Commit:
feat: :construction: Update testimonial component

Add a property for folder and remove property for imageurl, edit property for imageAlt and adapt component for properties

Commit:
fix: :bug: Fix the name of the property required from Astr.props

Change the variable name that gets the mainImageAlt value from props

Commit:
feat: :construction: Add Testimonials to Design Page

Implement component in design page to show the examples for both pages

Commit:
style: :lipstick: Fix styles for testimonial cases

Fix styles to separate differences between testimonial variations, using folder property as identifier

Commit:
style: :lipstick: Add calc for position of testimonial shape

Add a calc for height on testimonial shape for hypophosphatasia testimonial

HackMort commented on src/components/Testimonial.astro:
```diff
@@ -0,0 +1,26 @@
+<section class={`testimonial testimonial--${page}`}>
+    <picture class="testimonial__shape-container">
+        <source media="(min-width: 1024px)" srcset={`/assets/images/${page}/testimonial/testimonial-shape-desktop.svg`}>
+        <img class="testimonial__shape" src={`/assets/images/${page}/testimonial/testimonial-shape-mobile.svg`} alt=" ">
```
I think this would be better if instead of sending a Prop "page" maybe we can send shapeType or something like that this way we can just send "top" or "bottom" even better set the defualt to top and make the shapeType Prop optional. I would also move all shapeTypes vectors to the "vectors" folder and create a shapes subfolder so we can have them all live in the same place just different formats/styles.
```suggestion
        <img class="testimonial__shape" src={`/assets/images/${shapeType}/testimonial/testimonial-shape-mobile.svg`} alt=" ">

        Even better IMO
        <img class="testimonial__shape" src={`/assets/images/testimonial-shapes/${shapeType}/testimonial-shape-mobile.svg`} alt=" ">
```

HackMort commented on src/components/Testimonial.astro:
```diff
@@ -0,0 +1,26 @@
+        <img class="testimonial__shape" src={`/assets/images/${page}/testimonial/testimonial-shape-mobile.svg`} alt=" ">
+    </picture>
+    <figure class="testimonial__image-container">
+        <img src={`/assets/images/${page}/testimonial/testimonial.png`} alt={mainImageAlt} class="testimonial__image">
```
This page Prop makes sense, maybe we need to different ones, page and shapeType based on my previous comment.

HackMort commented on src/components/Testimonial.astro:
```diff
@@ -0,0 +1,26 @@
+        <img src={`/assets/images/${page}/testimonial/testimonial.png`} alt={mainImageAlt} class="testimonial__image">
+    </figure>
+    <article class="testimonial__content">
+        <blockquote class="testimonial__quote"><p>{quote}</p></blockquote>
```
We may need to add a Testimonial with more than 1 paragraph of text. So I recomend this to be a slot/children and if we need multiple paragraphs, we can just send them.

```suggestion
        <Testimonial
    authorInformation="Lynn Kohlmeier, MD, Adult Endocrinologist"
    authorDescription="Alexion-sponsored speaker"
    page = "home"
  >
        <p>Something</p>
        <p>Lorem ipsum.....</p>
  </Testimonial>

```

HackMort commented on src/styles/components/_testimonial.scss:
```diff
@@ -0,0 +1,224 @@
+.testimonial {
```
I like the custom properties idea for fake variables. Just please do not use PX us Rems, that why I send you guys the vs code extension to convert px to rems.

HackMort reviewed (changes requested):
Overall looks pretty good. Please see my comments and fix. 

Also I noticed that on iOS Devices there's a weird margin being applied to the picture element. I think this is caused by the mixin of flex-gap, but it can be "fixed" by directly set margin: 0 to the picture element, see screenshot below:

![image](https://user-images.githubusercontent.com/6466646/231576998-d25fa516-332a-436f-b8dd-674ed95f6dcf.png)

Commit:
Merge branch 'develop' of https://github.com/HackMort/strensiq-hcp into feature/testimonial-component

Commit:
fix: :bug: Set margin to 0 for picture element to fix iOS issue

Set margin with value 0 to fix an  issue on iOS devices in picture element

Commit:
style: :lipstick: Change px to rem in testimonial scss

Change CSS values on px to rem , except for font-sizes

Commit:
feat: :construction: Change position of Testimonial on Home

Change Testimonial component on home page accodring to design

Commit:
Merge branch 'develop' of https://github.com/HackMort/strensiq-hcp into feature/testimonial-component

Commit:
feat: :construction: Change Testimonial component to allow multiple paragraphs

Add a slot inside blockquote element to allow to use multiple pargraphs on Testimonial., change CSS for some elements like the quotes

Commit:
fix: :bug: Fix height of shape when add more than 2 paragraphs

Fix height set its value in percentage instead of rem to get the height of the container

Commit:
feat: :construction: Add a property called shapeType for Testimonial component

Add an optional property called shapeType, default value is top, to identify between the shapeType for the Testimonial and the page where could need changes or consume a mainImage, create required folders, move shape vectors to correct folder and change CSS

Commit:
Merge branch 'develop' of https://github.com/HackMort/strensiq-hcp into feature/testimonial-component

Commit:
fix: :fire: Remove extra paragraph from Testimonial in Home Page

Remove an extra paragraph from Testimonial in Home Page , just have to go on examples

Commit:
Merge branch 'develop' of https://github.com/HackMort/strensiq-hcp into feature/testimonial-component

Commit:
fix: :bug: Fix bug of shape image for top shape type

Fix the image URL for testimonial shape type

HackMort reviewed (approved):
Testimonials look great!
